lib/spack/spack/user_environment.py

```diff
@@ -102,6 +102,9 @@ def environment_modifications_for_specs(
 
     # Static environment changes (prefix inspections)
     for s in reversed(topo_ordered):
+        for p in getattr(s.package, "cmake_prefix_paths", []):
```

Comment:
Does it make sense to reuse get_cmake_prefix_path here?
 https://github.com/spack/spack/blob/e9e6eb613b752bb3e40f324000896b4c2668587f/lib/spack/spack/build_environment.py#L1068